test/Microsoft.Health.Dicom.Tests.Integration/Persistence/ExtendedQueryTagErrorStoreTests.cs

```diff
@@ -0,0 +1,172 @@
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Tests.Integration.Persistence
+{
+    public class ExtendedQueryTagErrorStoreTests : IClassFixture<SqlDataStoreTestsFixture>, IAsyncLifetime
+    {
+        private readonly IExtendedQueryTagErrorStore _extendedQueryTagErrorStore;
+        private readonly IExtendedQueryTagStore _extendedQueryTagStore;
+        private readonly IIndexDataStore _indexDataStore;
+        private readonly IIndexDataStoreTestHelper _testHelper;
+        private readonly DateTime _definedNow;
+        private long _watermark = 0;
+
+        public ExtendedQueryTagErrorStoreTests(SqlDataStoreTestsFixture fixture)
+        {
+            EnsureArg.IsNotNull(fixture, nameof(fixture));
+            _extendedQueryTagStore = EnsureArg.IsNotNull(fixture.ExtendedQueryTagStore, nameof(fixture.ExtendedQueryTagStore));
+            _extendedQueryTagErrorStore = EnsureArg.IsNotNull(fixture.ExtendedQueryTagErrorStore, nameof(fixture.ExtendedQueryTagErrorStore));
+            _indexDataStore = EnsureArg.IsNotNull(fixture.IndexDataStore, nameof(fixture.IndexDataStore));
+            _testHelper = EnsureArg.IsNotNull(fixture.TestHelper, nameof(fixture.TestHelper));
+            _definedNow = DateTime.UtcNow;
+
+        }
+        public async Task DisposeAsync()
+        {
+            await CleanupTagErrorsAsync();
+        }
+
+        private async Task CleanupTagErrorsAsync()
+        {
+            var tags = await _extendedQueryTagStore.GetExtendedQueryTagsAsync();
+            foreach (var tag in tags)
+            {
+                await _extendedQueryTagErrorStore.DeleteExtendedQueryTagErrorsAsync(tag.Path);
+                await _extendedQueryTagStore.DeleteExtendedQueryTagAsync(tag.Path, tag.VR);
+            }
+        }
+
+        private async Task<ExtendedQueryTagStoreEntry> CreateTagInStoreAsync(CancellationToken cancellationToken = default)
+        {
+            DicomTag tag1 = DicomTag.DeviceSerialNumber;
+            AddExtendedQueryTagEntry extendedQueryTagEntry1 = tag1.BuildAddExtendedQueryTagEntry();
+
+            // need to create instance too
+            DicomDataset dataset = CreateTestDicomDataset(TestUidGenerator.Generate(), TestUidGenerator.Generate(), TestUidGenerator.Generate());
+            _watermark = await _indexDataStore.CreateInstanceIndexAsync(dataset);
+
+            await _extendedQueryTagStore.AddExtendedQueryTagsAsync(new AddExtendedQueryTagEntry[] { extendedQueryTagEntry1 }, 128, ready: true, cancellationToken: cancellationToken);
+
+            var actualExtendedQueryTagEntries = await _extendedQueryTagStore.GetExtendedQueryTagsAsync(extendedQueryTagEntry1.Path);
+
+            Assert.Equal(actualExtendedQueryTagEntries[0].Path, extendedQueryTagEntry1.Path);
+            return actualExtendedQueryTagEntries[0];
+        }
+
+        private static DicomDataset CreateTestDicomDataset(string studyInstanceUid = null, string seriesInstanceUid = null, string sopInstanceUid = null)
```

Comment:
can I put this in test.Common?
or somewhere, since it is reused from IndexDataStoreTest

Reply:
Could we use https://github.com/microsoft/dicom-server/blob/83da5fa9f193320269198c272ea7d73a116358d5/tools/scale-testing/Common/Samples.cs#L222